ml_gobject0.c

```diff
@@ -40,6 +41,19 @@ value Val_GObject(GObject *obj) {
   return result;
 }
 
+value Val_string_option(char *s) {
+  CAMLparam0();
+  if (s) {
+    CAMLlocal2(ml_s, result);
+    ml_s = caml_copy_string(s);
+    result = caml_alloc(1, 0);
+    Store_field(result, 0, ml_s);
```

Comment:
Use `caml_alloc_some`?

Reply:
done. wasn't sure if we wanted to drop support for pre-4.12.